projects/js-packages/svelte-data-sync-client/src/initializeClient.ts

```diff
@@ -0,0 +1,106 @@
+import { z } from 'zod';
+import { API } from './API';
+import { API_Endpoint } from './Endpoint';
+import { SyncedStore } from './SyncedStore';
+import { ValidatedValue } from './types';
+
+export function getValidatedValue< T extends z.ZodSchema >(
+	namespace: string,
+	valueName: string,
+	valueSchema: T
+): ValidatedValue< T > {
+	const validator = z.object( {
+		value: valueSchema,
+		nonce: z.string(),
+	} );
+
+	/**
+	 * This function handles fetching nonces if they're there
+	 * But it shouldn't trigger errors if the nonce is missing.
+	 * Allowing the client application decide how to handle
+	 * missing nonces.
+	 *
+	 * This is useful, for example, if the `valueSchema` is set to something
+	 * like `z.string().optional()`. In this case, the application might be
+	 * okay with a missing value, so a missing nonce would trigger an unexpected
+	 * error.
+	 */
+	const source = window[ namespace ][ valueName ]
+		? window[ namespace ][ valueName ]
+		: { value: undefined, nonce: '' };
+
+	const result = validator.parse( source );
+
+	return result as ValidatedValue< T >;
+}
+
+// `window.[name].rest_api` is a special value that's required for the API to work.
+// It's populated by the `wp-js-data-sync` package and contains the REST API endpoint and nonce.
+function setupRestApi( namespace: string ) {
+	const api = new API();
+	try {
+		const { value, nonce } = getValidatedValue( namespace, 'rest_api', z.string().url() );
+		api.initialize( value, nonce );
+	} catch ( e ) {
```

Comment:
On second thought, by doing this you would have to refactor some code like [here](https://github.com/Automattic/jetpack/blob/e17cbd6513ceddd6977608d35b382ad6b2cf1ab3/projects/js-packages/svelte-data-sync-client/src/initializeClient.ts#L39-L58) so it might not be a good idea?